tests/test_acasclient.py

```diff
@@ -2296,6 +2296,144 @@ def test046_cmpdreg_admin_crud(self):
         self.client.delete_cmpdreg_vendor(vendor['id'])
         
 
+    def test_047_load_sdf_with_salts(self):
+        """
+        Test to Load SDF File w/ Salts to Determine Process
+        """
+        test_047_load_sdf_with_salts_file = Path(__file__).resolve().parent.\
+            joinpath('test_acasclient', 'test_047_register_sdf_with_salts.sdf')
+        mappings = [
+            {
+                "dbProperty": "Parent Corp Name",
+                "defaultVal": None,
+                "required": False,
+                "sdfProperty": "Corporate ID"
+            },
+            {
+                "dbProperty": "Lot Chemist",
+                "defaultVal": "bob",
+                "required": True,
+                "sdfProperty": "Lot Scientist"
+            },
+            {
+                "dbProperty": "Project",
+                "defaultVal": "PROJ-00000001",
+                "required": True,
+                "sdfProperty": "Project Code Name"
+            },
+            {
+                "dbProperty": "Parent Stereo Category",
+                "defaultVal": "unknown",
+                "required": True,
+                "sdfProperty": None
+            },
+            {
+                "dbProperty": "Lot Salt Abbrev",
+                "defaultVal": None,
+                "required": False,
+                "sdfProperty": "Lot Salt Name"
+            },
+            {
+                "dbProperty": "Lot Salt Equivalents",
+                "defaultVal": None,
+                "required": False,
+                "sdfProperty": "Lot Salt Equivalents"
+            }
+        ]
+
+        SALT_ABBREV = 'HCl'
+        SALT_MOL = "\n  Ketcher 05182214202D 1   1.00000     0.00000     0\n\n  1  0  0     1  0            999 V2000\n    6.9500   -4.3250    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0\nM  END\n"
+        # Get Salt Abbrevs. Treat salts separately since they are not a standard codetable
+        salts = self.client.get_salts()
+        # Create Salt Abbrev
+        if SALT_ABBREV.lower() not in [s['abbrev'].lower() for s in salts]:
+            salt = self.client.create_salt(abbrev=SALT_ABBREV, name=SALT_ABBREV, mol_structure=SALT_MOL)
+            self.assertIsNotNone(salt.get('id'))
+
+        response = self.client.register_sdf(test_047_load_sdf_with_salts_file, "bob",
+                                            mappings)
+        print(response)
+        self.assertIn('report_files', response)
+        self.assertIn('summary', response)
```

Comment:
I was expecting a few more checks here. What about checking that the error is thrown x number of times and that the records you expect to be successful were successfully registered?

Reply:
Implemented